mlflow/tracing/processor/mlflow.py

```diff
@@ -70,6 +71,15 @@ def _start_trace(self, span: OTelSpan) -> TraceInfo:
         experiment_id = (
             get_otel_attribute(span, SpanAttributeKey.EXPERIMENT_ID) or _get_experiment_id()
         )
+        if experiment_id == DEFAULT_EXPERIMENT_ID:
+            _logger.warning(
+                "Creating a trace within the default experiment with id "
+                f"'{DEFAULT_EXPERIMENT_ID}'. However, it is strongly recommended to set a "
+                "specific experiment when using tracing, because the default experiment can "
+                "become a monolithic huge dump with thousands of traces and negatively "
+                "impact storage performance. To avoid this, set the experiment for "
+                "your environment using `mlflow.set_experiment()` API."
```

Comment:
```suggestion
                "Creating a trace within the default experiment with id "
                f"'{DEFAULT_EXPERIMENT_ID}'. It is strongly recommended to not use "
                "the default experiment to log traces due to ambiguous search results and "
                "probable performance issues over time due to directory table listing performance "
                "degradation with high volumes of directories within a specific path. "
                "To avoid performance and disambiguation issues, set the experiment for "
                "your environment using `mlflow.set_experiment()` API."
```
Adding a call-out to the impacts of the usability of search_traces if loads of models' traces are all dumped in a single experiment :)